### DISCUSSION TITLE
Disable Native Application Menu

### DISCUSSION THREAD

**Discussion by teocomi** [AvaloniaUI/Avalonia#6565] [Q&A]
Hello,

I'm trying to find a way to disable the native application menu bar showing in OSX, is it possible?

Some context:

I'm launching my Avalonia App as an addin inside another software, but when it does on Mac, it replaces the host application menu, and this is undesirable.

Thanks!

**Comment by maxkatz6** [comment]
@danwalmsley

**Reply by teocomi** [reply]
Hey @maxkatz6 @danwalmsley any pointers on this would be much appreciated, I spent a long time through the source code but could not find a solution! Txs

**Comment by kekekeks** [comment]
If one were to add a "disable menu" flag, it would be required to add checks at least in
https://github.com/AvaloniaUI/Avalonia/blob/0dea06353102d71588c6687ddb14392e33d9e45f/native/Avalonia.Native/src/OSX/app.mm#L31
https://github.com/AvaloniaUI/Avalonia/blob/27000e1cc9c8e722074ddcc6f6f79785654287f7/native/Avalonia.Native/src/OSX/menu.mm#L465
https://github.com/AvaloniaUI/Avalonia/blob/864e071a78384647f19326cc2d4d1c2e906b5179/native/Avalonia.Native/src/OSX/window.mm#L2103


and probably some other places and then fix the managed part to not register/expose INativeMenuExporter instances if said flag is set.

**Reply by teocomi** [reply]
Thanks @kekekeks , I'm afraid that's too much out of my comfort zone!

Do you think I could achieve something similar with code below by accessing the `NSApplicaiton` and `NSMenu`?
I've also included a video of the issue below to better describe it, as you can see when the Avalonia app launches it takes over the previous Rhino app bar. I'd simply want the app to not do that :)

```
using Avalonia;
using Avalonia.Controls.ApplicationLifetimes;
using Avalonia.Markup.Xaml;
using DesktopUI2.ViewModels;
using DesktopUI2.Views;
using MonoMac.AppKit;
using MonoMac.Foundation;

namespace DesktopUI2
{
  public class App : Application
  {
    public override void Initialize()
    {
      AvaloniaXamlLoader.Load(this);
    }

    public override void OnFrameworkInitializationCompleted()
    {
      if (ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
      {
        desktop.MainWindow = new MainWindow
        {
          DataContext = new MainWindowViewModel(),
        };
      }

      NSApplication.Init();
      NSApplication.SharedApplication.Delegate = new OSXAppDelegate();

      base.OnFrameworkInitializationCompleted();
    }
  }

  public class OSXAppDelegate : NSApplicationDelegate
  {
    public OSXAppDelegate()
    {
      AvaloniaLocator.CurrentMutable.Bind<OSXAppDelegate>().ToConstant(this);
      NSMenu.MenuBarVisible = false;//hides the app menu bar entirely, not what I need  
    }
  }
}
```

Sorry for the handheld video, but could not record the app bar otherwise!

https://user-images.githubusercontent.com/2679513/136053797-67bf8017-89e4-4320-9abb-e2f0de74bceb.MOV
